### PR TITLE
[samplecode][3/3]Implement Stream.Client sample code

### PR DESCRIPTION
**Pull request by summer-ji-eng**
```   
* try (EchoClient echoClient = EchoClient.create()) {
   *   ApiStreamObserver<EchoRequest> responseObserver =
   *       new ApiStreamObserver<EchoRequest>() {
   *         @Override
   *         public void onNext(EchoResponse response) {
   *           // Do something when receive a response.
   *         }
   *
   *         @Override
   *         public void onError(Throwable t) {
   *           // Add error-handling
   *         }
   *
   *         @Override
   *         public void onCompleted() {
   *           // Do something when complete.
   *         }
   *       };
   *   ApiStreamObserver<EchoRequest> requestObserver =
   *       echoClient.collect().clientStreamCall(responseObserver);
   *   EchoRequest request =
   *       EchoRequest.newBuilder()
   *           .setName(FoobarName.ofProjectFoobarName("[PROJECT]", "[FOOBAR]").toString())
   *           .setParent(FoobarName.ofProjectFoobarName("[PROJECT]", "[FOOBAR]").toString())
   *           .setFoobar(Foobar.newBuilder().build())
   *           .build();
   *   requestObserver.onNext(request);
   * }
```

### CODE REVIEW

File: src/main/java/com/google/api/generator/engine/ast/JavaDocComment.java [hunk 1]
```diff
@@ -92,7 +92,7 @@ public Builder addSampleCode(String sampleCode) {
       Arrays.stream(sampleCode.split("\\r?\\n"))
           .forEach(
               line -> {
-                componentsList.add(line);
+                componentsList.add(line.replace("@Override", "{@literal @}Override"));
```
Comment (miraleung) [inline src/main/java/com/google/api/generator/engine/ast/JavaDocComment.java:95]: Is this meant to be temporary? Shouldn't we handle `@` in general, for annotations that are not `@Override`?

File: src/main/java/com/google/api/generator/gapic/composer/samplecode/ServiceClientSampleCodeComposer.java [hunk 2]
```diff
@@ -562,6 +565,10 @@ public static String composeStreamCallableMethodHeaderSampleCode(
     } else if (method.stream().equals(Stream.BIDI)) {
       bodyStatements.addAll(
           composeStreamBidiSampleCodeBodyStatements(method, clientVarExpr, requestAssignmentExpr));
+    } else {
```
Comment (miraleung) [inline src/main/java/com/google/api/generator/gapic/composer/samplecode/ServiceClientSampleCodeComposer.java:568]: What if someone passes `Stream.NONE` into this method? Can we add an early return earlier, and/or an `else if` here?

File: src/main/java/com/google/api/generator/gapic/composer/samplecode/ServiceClientSampleCodeComposer.java [hunk 3]
```diff
@@ -842,6 +849,131 @@ public static String composeStreamCallableMethodHeaderSampleCode(
...
+    TypeNode responseObserverType =
+        TypeNode.withReference(
+            ConcreteReference.builder()
+                .setClazz(ApiStreamObserver.class)
+                .setGenerics(method.inputType().reference())
+                .build());
+    VariableExpr responseObserverVarExpr =
+        VariableExpr.withVariable(
+            Variable.builder().setName("responseObserver").setType(responseObserverType).build());
+    // Create Anonymous Class contains methods: onNext, onError, onCompleted.
```
Comment (miraleung) [inline src/main/java/com/google/api/generator/gapic/composer/samplecode/ServiceClientSampleCodeComposer.java:868]: Nit: Grammar and regular casing rules apply in comments. "Create anonymous class" or "Create an anonymous class for ApiStreamObserver that contains the methods onNext, onError, and onCompleted."

**Review by miraleung — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/api/generator/gapic/composer/samplecode/ServiceClientSampleCodeComposer.java (hunk 2 above)
Comment (summer-ji-eng) [reply src/main/java/com/google/api/generator/gapic/composer/samplecode/ServiceClientSampleCodeComposer.java:568]: The caller is from `ServiceClientClassComposer` ([code](https://github.com/googleapis/gapic-generator-java/blob/stream_client_sample_code/src/main/java/com/google/api/generator/gapic/composer/ServiceClientClassComposer.java#L808-L816)),
```    if (!method.stream().equals(Stream.NONE)) {
      sampleCodeOpt =
          Optional.of(
              ServiceClientSampleCodeComposer.composeStreamCallableMethodHeaderSampleCode(
                  method,
                  typeStore.get(ClassNames.getServiceClientClassName(service)),
                  resourceNames,
                  messageTypes));
    }
```
Added `else if` here.

File: src/main/java/com/google/api/generator/gapic/composer/samplecode/ServiceClientSampleCodeComposer.java (hunk 3 above)
Comment (summer-ji-eng) [reply src/main/java/com/google/api/generator/gapic/composer/samplecode/ServiceClientSampleCodeComposer.java:868]: Agree. Thanks for correcting.

File: src/main/java/com/google/api/generator/engine/ast/JavaDocComment.java (hunk 1 above)
Comment (summer-ji-eng) [reply src/main/java/com/google/api/generator/engine/ast/JavaDocComment.java:95]: It is temporary solution.

Comment (miraleung) [reply src/main/java/com/google/api/generator/engine/ast/JavaDocComment.java:95]: Can we add a TODO here?

File: src/main/java/com/google/api/generator/gapic/composer/samplecode/ServiceClientSampleCodeComposer.java [hunk 4]
```diff
@@ -843,6 +850,131 @@ public static String composeStreamCallableMethodHeaderSampleCode(
...
+            .setArguments(
+                VariableExpr.builder()
+                    .setVariable(
+                        Variable.builder().setName("response").setType(method.outputType()).build())
+                    .setIsDecl(true)
+                    .build())
+            .setBody(
+                Arrays.asList(
+                    CommentStatement.withComment(
+                        LineComment.withComment("Do something when receive a response."))))
```
Comment (miraleung) [inline src/main/java/com/google/api/generator/gapic/composer/samplecode/ServiceClientSampleCodeComposer.java:884]: Grammar in comments (might have been wrong in the monolith to begin with): "Do something when a response is received."

**Review by miraleung — APPROVED** [review]
LGTM with comments addressed.

File: src/main/java/com/google/api/generator/engine/ast/JavaDocComment.java (hunk 1 above)
Comment (summer-ji-eng) [reply src/main/java/com/google/api/generator/engine/ast/JavaDocComment.java:95]: I have problem here. If we handle  `@` in  annotations `JavaWriteVistor`, the `google-java-formatter` will complain an error `com.google.googlejavaformat.java.FormatterException: 3:13: error: <identifier> expected`

`JavaWriterVisitor` looks like 
```
  @Override
  public void visit(AnnotationNode annotation) {
    if (!annotation.escapeCharacterAt()) {
      buffer.append(AT);
    } else {
      leftBrace();
      buffer.append(AT);
      buffer.append(LITERAL);
      space();
      buffer.append(AT);
      rightBrace();
    }
    annotation.type().accept(this);
    if (annotation.descriptionExpr() != null) {
      leftParen();
      annotation.descriptionExpr().accept(this);
      rightParen();
    }
    newline();
  }
```

In this case, we have to format the code first, and then escape the character `@`. What do you think? Thank @miraleung

Comment (summer-ji-eng) [reply src/main/java/com/google/api/generator/engine/ast/JavaDocComment.java:95]: Will add `todo` to mention it is the temporary solution, and will discuss the long-term solution off channel.

File: src/main/java/com/google/api/generator/gapic/composer/samplecode/ServiceClientSampleCodeComposer.java (hunk 4 above)
Comment (summer-ji-eng) [reply src/main/java/com/google/api/generator/gapic/composer/samplecode/ServiceClientSampleCodeComposer.java]: Done.

**Pull request merged**